urlcdab7d8364_jrwalsh_CycTools_tgz-pJ8-edu_iastate_cyctools_MainJ8/src/edu/iastate/cyctools/CycDataBaseAccess.java

```diff
@@ -0,0 +1,687 @@
+				e.printStackTrace();
+			}
+		}
+	}
+	
+	public Frame updateLocalFrame(String frameID, ArrayList<AbstractFrameEdit> frameUpdates) {
+		Frame frame = null;
+		try {
+			frame = Frame.load(conn, frameID);
+			frame.update();
+			for (AbstractFrameEdit frameUpdate : frameUpdates) {
+				frameUpdate.commitLocal(frame);
+			}
+		} catch (PtoolsErrorException e) {
+			e.printStackTrace();
+		}
+		return frame;
+	}
+	
+	public DefaultMutableTreeNode frameToTree(String frameID) {
+ 		Frame frame = null;
+ 		try {
+			frame = Frame.load(conn, frameID);
+		} catch (PtoolsErrorException e) {
+			e.printStackTrace();
+			return null;
+		}
+ 		
+ 		DefaultMutableTreeNode root = null;
+ 		try {
+ 			root = createNodesFromFrame(new DefaultMutableTreeNode(frame.getLocalID()), frame);
+		} catch (PtoolsErrorException e) {
+			e.printStackTrace();
+		}
+ 		
+ 		return root;
+ 	}
+ 	
+	private DefaultMutableTreeNode createNodesFromFrame(DefaultMutableTreeNode top, Frame frame) throws PtoolsErrorException {
+ 		// Get slot names
+		HashMap<String, ArrayList> slots = frame.getSlots();
+ 		for (String slotLabel : slots.keySet()) {
+			DefaultMutableTreeNode slotNode = new DefaultMutableTreeNode(slotLabel);
+ 			// Get slot values
+			ArrayList<String> slotValues = new ArrayList<String>();
+ 			for (Object slotValue : conn.getSlotValues(frame.getLocalID(), slotLabel)) {
+ 				if (slotValue instanceof String) slotValues.add((String) slotValue);
+ 				else slotValues.add(JavacycConnection.ArrayList2LispList((ArrayList) slotValue));
+ 			}
+ 			for (String slotValue : slotValues) {
+ 					DefaultMutableTreeNode slotValueNode = new DefaultMutableTreeNode(slotValue);
+ 	 				// Get annotation names
+ 	 				ArrayList<String> annotLabels = frame.getAllAnnotLabels(slotLabel, slotValue);
+ 	 				for (String annotLabel : annotLabels) {
+ 	 					DefaultMutableTreeNode annotLabelNode = new DefaultMutableTreeNode(annotLabel);
+ 	 					// Get annotation values
+ 	 					ArrayList<String> annotValues = (ArrayList<String>) frame.getAnnotations(slotLabel, slotValue, annotLabel);
+ 	 					for (String annotValue: annotValues) {
+ 	 						DefaultMutableTreeNode annotValueNode = new DefaultMutableTreeNode(annotValue);
+ 	 						annotLabelNode.add(annotValueNode);
+ 	 					}
+ 	 					slotValueNode.add(annotLabelNode);
+ 	 				}
+ 	 				slotNode.add(slotValueNode);
+ 			}
+ 			top.add(slotNode);
+ 		}
+ 		return top;
+ 	}
+	
+	public DefaultTableModel getSearchResultsTable(String text, String type) {
+		progressMonitor = new ProgressMonitor(DefaultController.mainJFrame, "Running a Long Task", "", 0, 100);
+		progressMonitor.setMillisToDecideToPopup(0);
+		progressMonitor.setMinimum(0);
+		progressMonitor.setProgress(0);
+		task = new GetSearchResultsTableTask(text, type);
+		task.addPropertyChangeListener(this);
+		task.execute();
+		
+		try {
+			if (task.get() != null) {
+				return (DefaultTableModel)task.get();
+			}
+		} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)